src/hub/action_form.ts

```diff
@@ -14,7 +20,9 @@ export interface ActionFormField {
   label?: string
   description?: string
   default?: string
-  type?: "string" | "textarea" | "select"
+  type?: "string" | "textarea" | "select" | "set_params_link" | "oauth_link"
   options?: { name: string, label: string }[]
   required?: boolean
+  oauth_url?: string
+  settable_params?: string[]
```

Comment:
If these are only respected when `type` has a particular value, we should encode this into the `ActionFormField` type so the type system will prevent you from specifying them on a `string`.